src/CommunityToolkit.Maui.MediaElement/Extensions/SubtitleExtensions.android.cs

```diff
@@ -0,0 +1,242 @@
+
+	void InitializeTextBlock()
+	{
+		var (currentActivity, _, _, _) = VerifyAndRetrieveCurrentWindowResources();
+		var activity = currentActivity;
+		subtitleView = new(activity.ApplicationContext)
+		{
+			Text = string.Empty,
+			HorizontalScrollBarEnabled = false,
+			VerticalScrollBarEnabled = false,
+			TextAlignment = Android.Views.TextAlignment.Center,
+			Visibility = Android.Views.ViewStates.Gone,
+			LayoutParameters = subtitleLayout
+		};
+		subtitleView.SetBackgroundColor(Android.Graphics.Color.Argb(150, 0, 0, 0));
+		subtitleView.SetTextColor(Android.Graphics.Color.White);
+		subtitleView.SetPaddingRelative(10, 10, 10, 20);
+	}
+
+	static (Activity CurrentActivity, Android.Views.Window CurrentWindow, Resources CurrentWindowResources, Configuration CurrentWindowConfiguration) VerifyAndRetrieveCurrentWindowResources()
+	{
+		// Ensure current activity and window are available
+		if (Platform.CurrentActivity is not Activity currentActivity)
+		{
+			throw new InvalidOperationException("CurrentActivity cannot be null when the FullScreen button is tapped");
+		}
+		if (currentActivity.Window is not Android.Views.Window currentWindow)
+		{
+			throw new InvalidOperationException("CurrentActivity Window cannot be null when the FullScreen button is tapped");
+		}
+
+		if (currentActivity.Resources is not Resources currentResources)
+		{
+			throw new InvalidOperationException("CurrentActivity Resources cannot be null when the FullScreen button is tapped");
+		}
+
+		if (currentResources.Configuration is not Configuration configuration)
+		{
+			throw new InvalidOperationException("CurrentActivity Configuration cannot be null when the FullScreen button is tapped");
+		}
+
+		return (currentActivity, currentWindow, currentResources, configuration);
+	}
+
+	void OnWindowStatusChanged(object? sender, WindowsEventArgs e)
+	{
+		ArgumentNullException.ThrowIfNull(subtitleView);
+
+		// If the subtitle URL is empty do nothing
+		if (string.IsNullOrEmpty(mediaElement?.SubtitleUrl))
+		{
+			return;
+		}
+		var (_, currentWindow, _, _) = VerifyAndRetrieveCurrentWindowResources();
+		if (currentWindow?.DecorView is not ViewGroup viewGroup)
+		{
+			return;
+		}
+		if (viewGroup.Parent is not ViewGroup parent)
+		{
+			return;
+		}
+		switch (e.data)
+		{
+			case true:
+				viewGroup.RemoveView(subtitleView);
+				InitializeTextBlock();
+				parent.AddView(subtitleView);
+				break;
+			case false:
+				parent.RemoveView(subtitleView);
+				InitializeTextBlock();
+				viewGroup.AddView(subtitleView);
+				break;
+		}
+	}
+
+	protected virtual void Dispose(bool disposing)
+	{
+		if (!disposedValue)
+		{
+			if (timer is not null)
+			{
+				timer.Stop();
+				timer.Elapsed -= UpdateSubtitle;
+			}
+
+			if (disposing)
+			{
+				httpClient.Dispose();
+				timer?.Dispose();
+				subtitleView?.Dispose();
+			}
+
+			timer = null;
+			subtitleView = null;
+			disposedValue = true;
+		}
+	}
+
+	~SubtitleExtensions()
+	{
+	     Dispose(disposing: false);
```

Comment:
I believe we don't need this, all the disposable resources hold by this class are Managed, that means GC will not collect this class if we don't call dispose manually. Friendly reminder that objects that implement a finalizer will live longer